download_scripts/change_dataset_preprocessing.py

```diff
@@ -0,0 +1,108 @@
+from typing import Optional, Dict, List
+
+import pandas as pd
+import re
+
+normalise_dataset_name_regex = re.compile(r"^(?:/gpfswork/rech/six/uty16tp/dataset/tokenization/)?(bigscience-catalogue-lm-data/[^/]+)(?:/data)?$")
+def get_dedup_args(row: Dict) -> List[str]:
+    ds_name = normalise_dataset_name_regex.match(row["dataset_name"]).group(1)
+    if "pseudocrawl" in ds_name:
+        list_of_dedups = ["dedup_document_on_url", "dedup_document", "dedup_pseudocrawl_newspapers"]
+    else:
+        list_of_dedups = ["dedup_document", "dedup_template_soft"]
+
+    if all(black_list_ds not in ds_name for black_list_ds in ["open_subtitles", "europarl", "uncorpus"]):
+        list_of_dedups += ["dedup_template_soft"]
+
+    list_of_dedups += ["filter_remove_empty_docs"]
+    return list_of_dedups
+
+
+language_regex = re.compile(r"^(?:/gpfswork/rech/six/uty16tp/dataset/tokenization/)?bigscience-catalogue-lm-data/lm_([^_]+)_.*(?:/data)?$")
+def get_language(dataset_name: str):
+    lang_candidate = language_regex.match(dataset_name).group(1)
+
+    # Normalise chinese languages, so that we only consider simplified and traditional chinese as the two chinese languages
+    if lang_candidate in ["zh", "zhs", "zh-cn"]:
+        lang_candidate = "zhs"
+    elif lang_candidate in ["zht", "zh-tw"]:
+        lang_candidate = "zht"
+    else:
+        assert lang_candidate[:2] != "zh"
+
+
+
+    return lang_candidate
```

Comment:
is there a reason not to map the chinese to just `zh`. I could use the same function for sentence splitting.

Reply:
I'm factorise the code to be in `clean_helpers/utils`